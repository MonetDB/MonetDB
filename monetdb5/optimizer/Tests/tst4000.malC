# The 4000 series deals with strategic optimizer constructs
# common subexpression evaluation
function foo();
b:= bat.new(:int);
c:= bat.new(:int);       # identical until insert takes place
d:= algebra.select(b,nil:bat[:oid],0,100,true,true,false);
e:= algebra.select(b,nil:bat[:oid],0,100,true,true,false);    # mapped onto one instruction
k1:= 24;
k2:= 27;
l:= k1+k2;
l2:= k1+k2;             # mapped to l
l3:= l2+k1;             # replace the alias
end;
optimizer.commonTerms("user","foo");
optimizer.aliases("user","foo");
mdb.List("user","foo");

