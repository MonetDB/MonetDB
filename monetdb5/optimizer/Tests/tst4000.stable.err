stderr of test 'tst4000` in directory 'monetdb5/optimizer` itself:


# 15:37:23 >  
# 15:37:23 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/Linux/etc/MonetDB5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5/Linux/var/MonetDB5/log" --set mapi_port=47817 --set sql_port=51746 --set monet_prompt= --trace "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --dbname=mTests_src_mal  tst4000.mal </dev/null
# 15:37:23 >  


# 15:37:23 >  
# 15:37:23 >  Done.
# 15:37:23 >  

