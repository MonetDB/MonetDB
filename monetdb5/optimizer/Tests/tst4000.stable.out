stdout of test 'tst4000` in directory 'monetdb5/optimizer` itself:


# 23:52:51 >  
# 23:52:51 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/log" --set mapi_port=39352 --set sql_port=45171 --set xquery_port=52900 --set monet_prompt= --trace --dbname=mTests_src_optimizer  tst4000.mal </dev/null
# 23:52:51 >  

# MonetDB server v5.7.0, based on kernel v1.25.0
# Serving database 'mTests_src_optimizer', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:33343/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-8837/.s.monetdb.33343
# MonetDB/GIS module loaded
# Start processing logs sql/sql_logs version 52200
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/R   module loaded


# 06:40:49 >  
# 06:40:49 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-8837" "--port=33343"
# 06:40:49 >  

function user.foo():void;               	#[0] (0)  0 
    b:bat[:int] := bat.new(:int);       	#[1] (0) CMDBATnew 1 <- 2 
    c:bat[:int] := bat.new(:int);       	#[2] (0) CMDBATnew 3 <- 2 
# identical until insert takes place 
    d:bat[:oid] := algebra.select(b:bat[:int], nil:bat[:oid], 0:int, 100:int, true:bit, true:bit, false:bit);	#[4] (0) ALGselect2 5 <- 1 6 7 8 9 10 11 
# mapped onto one instruction 
    l:int := calc.+(24:int, 27:int);	#[9] (0) CMDvarADDsignal 19 <- 16 18 
# mapped to l 
    l3:int := calc.+(l:int, 24:int);	#[12] (0) CMDvarADDsignal 22 <- 19 16 
# replace the alias 
end user.foo;	#[14] (0)  
#commonTerms          actions= 2 time=5 usec 
#aliases              actions= 4 time=3 usec 

# 23:52:51 >  
# 23:52:51 >  Done.
# 23:52:51 >  

